clients/spark/core/src/main/hadoop2/scala/io/treeverse/clients/conditional/S3ClientBuilder.scala

```diff
@@ -1,19 +1,28 @@
 package io.treeverse.clients.conditional
 
-import com.amazonaws.ClientConfiguration
+import com.amazonaws.retry.PredefinedRetryPolicies.SDKDefaultRetryCondition
+import com.amazonaws.retry.RetryPolicy
 import com.amazonaws.services.s3.{AmazonS3, AmazonS3ClientBuilder}
+import com.amazonaws.{AmazonClientException, AmazonWebServiceRequest, ClientConfiguration, SdkClientException}
 import io.treeverse.clients.StorageUtils.S3.createAndValidateS3Client
 import org.apache.hadoop.conf.Configuration
 import org.slf4j.{Logger, LoggerFactory}
 
+class S3RetryCondition extends SDKDefaultRetryCondition {
+  override def shouldRetry(originalRequest: AmazonWebServiceRequest, exception: AmazonClientException, retriesAttempted: Int): Boolean =  {
+    super.shouldRetry(originalRequest, exception, retriesAttempted) || exception.isInstanceOf[SdkClientException]
```

Comment:
I don't understand, I would expect to retry only if we've not retried too many times _and_ it's an SDK client exception:
```suggestion
    super.shouldRetry(originalRequest, exception, retriesAttempted) && exception.isInstanceOf[SdkClientException]
```

Reply:
The suggestion would eliminate the default retry policy behavior that `SDKDefaultRetryCondition` provides.
I would want the default behavior OR the case that that's an `SdkClientException`